src/bin/inx-chronicle/api/stardust/poi/merkle_proof.rs

```diff
@@ -81,10 +85,23 @@ impl MerkleProof {
         self.compute_audit_path(&data, index)
     }
 
-    /// Calculates the "Merkle Audit Path"
+    /// Recursively computes the "Merkle Audit Path" for a certain `BlockId` that is given by its index in a list of
+    /// ordered and unique `BlockId`s.
+    ///
+    /// Given `n` block ids, the algorithm deterministically selects a `mid` (largest power of two
```

Comment:
When I suggested adding this kind of detailed comment, I didn't mean in the docs...This is for us, which is why I just used regular comments.

Reply:
I really cannot understand your reasoning here.

Reply:
that's unfortunate :see_no_evil: 

Then I try again to make my point... I don't think that a single line of this function needs explanation. What you really want is an explanation of how the overall algorithm works...at least that's what I understood from our discussions. Hence, IMHO the explanation should be outside the method body be it as a doc comment or a normal comment.